packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,241 @@
+
+```html
+<nimble-table>
+    <nimble-table-column-text
+        operational-data-field-name="progress"
+        field-name="formattedProgress"
+    >
+        Progress
+    <nimble-table-column-text>`
+</nimble-table>
+```
+
+```ts
+const originalData = [{ progress: 0.1 }, { progress: 0.2 }];
+const tableData = originalData.map(x => {
+    progress: x.progress;
+    formattedProgress: x ? `${100 * x.progress}%` : undefined;
+});
+table.data = tableData;
+```
+
+**Pros:**
+
+-   formatted data is specified up front, guaranteeing fast scroll performance
+-   powerful; clients can format data however they want, including via browser APIs which are i18n-friendly or via server-side logic
+
+**Cons:**
+
+-   Increased memory usage and data update time from clients pre-populating data with field for each formatted column
+-   Added complexity of writing procedural code even for simple formatting use cases
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+**Implementation Cost:**
+
+-   Exposing `operational-data-field-name` to be set by client code rather than column definition
+
+#### Alternative 2: Client specifies formatting function
+
+When configuring a column, clients could provide a callback function that converts data of any supported type into a formatted string.
+
+There isn't a good way to set a function as an attribute value on a column, so the function would be specified in JS code. One possible mechanism would be for clients to override an abstract base class and register a new column type:
+
+```html
+<nimble-table>
+    <my-app-progress-column
+        field-name="progress"
+    >
+        Progress
+    <my-app-progress-column>`
+</nimble-table>
+```
+
+```ts
+class MyAppProgressColumn : NimbleFormattedTextColumnBase<number> {
+    public override format(value: number) : string {
+        return `${100 * value}%`;
+    }
+
+    public override shouldUsePlaceholder(value: number | undefined) : boolean {
+        return value === undefined;
+    }
+}
+
+MyAppProgressColumn.registerColumn('my-app-progress-column');
+```
+
+Some of this is prototyped in the [number-column-prototype branch](https://github.com/ni/nimble/compare/main...number-column-prototype?expand=1).
+
+Other variants of this idea include:
+
+1. setting the formatting function as a property on a column element.
+2. setting an [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) object as a property on a column element.
```

Comment:
Note that this could necessitate providing some translation between something like the .NET [`NumberFormatInfo` object](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.numberformatinfo?view=net-7.0), and `Intl.NumberFormat`.

Reply:
After discussion, the current recommendation is that Blazor clients would provide this configuration in JS. I clarifed this in the Cons section.